### ISSUE TITLE
[Bug]: Specific website times out in Playwright

### ISSUE THREAD

**Issue by huaracheguarache** [microsoft/playwright-python#2486]
### Version

1.45.0

### Steps to reproduce

Example steps:
1. Clone my repo at https://github.com/huaracheguarache/playwright_bug
2. Move into cloned repository directory.
3. Run: `podman build -t minimal_example -f Containerfile .`
4. Run: `podman run --rm --name minimal_example minimal_example:latest`
5. You should see the following be printed:

```
Firefox timed out!
Chromium timed out!
Webkit timed out!
```

### Expected behavior

Playwright loads website without timing out.

### Actual behavior

Playwright times out while trying to load the website.

### Additional context

I have tested manually opening this website in my Firefox browser on my Fedora system, and it is able to load without issues. I initially suspected that this issue could have been due to the server rejecting requests from headless browsers, but I think I was able to show that this is not the case with the following steps:

* I was able to manually load the webpage with Google Chrome which I ran with the `--enable-automation` flag. That flag sets the [webdriver property](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/webdriver) to true.
* I have made sure to set fake user agents in the Python script.

The website the script tries to access is a Norwegian municipality website: https://www.bamble.kommune.no/

As I manually loaded the website in Firefox I noticed in the network tab that a few resources are failing to load:

![Screenshot from 2024-07-14 19-52-53](https://github.com/user-attachments/assets/02b0887f-0f6a-448d-b7a4-821647a22768)

I don't know whether this could be the reason behind Playwright timing out while trying to load the site, but I thought it would be worth mentioning it.

The Podman container runs the following Python script:

```python
from playwright.sync_api import sync_playwright
from playwright.sync_api import TimeoutError as PlaywrightTimeoutError

with sync_playwright() as p:
    browser = p.firefox.launch()
    context = browser.new_context(user_agent='Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0')
    page = context.new_page()
    try:
        page.goto("https://www.bamble.kommune.no/", timeout=30000)
    except PlaywrightTimeoutError:
        print('Firefox timed out!')
    else:
        print('Firefox succeeded!')
    finally:
        context.close()
        browser.close()

    browser = p.chromium.launch()
    context = browser.new_context(user_agent='Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)'
                                             ' Chrome/126.0.0.0 Safari/537.36')
    page = context.new_page()
    try:
        page.goto("https://www.bamble.kommune.no/", timeout=30000)
    except PlaywrightTimeoutError:
        print('Chromium timed out!')
    else:
        print('Chromium succeeded!')
    finally:
        context.close()
        browser.close()

    browser = p.webkit.launch()
    context = browser.new_context(user_agent='Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15'
                                             ' (KHTML, like Gecko) Version/17.5 Safari/605.1.15')
    page = context.new_page()
    try:
        page.goto("https://www.bamble.kommune.no/", timeout=30000)
    except PlaywrightTimeoutError:
        print('Webkit timed out!')
    else:
        print('Webkit succeeded!')
    finally:
        context.close()
        browser.close()
```

### Environment

```Text
- Operating System: Fedora 40 Silverblue
- CPU: x86_64
- Browser: Chromium, Firefox, WebKit
- Python Version: 3.12
- Other info: I'm using a Podman container.
```

**Comment by mxschmitt** [comment]
I'm unfortunately not familiar with Podman, but I think its similar to Docker under the hood, so I ran your image using the Docker CLI. For me it gave:

```
Firefox succeeded!
Chromium succeeded!
Webkit succeeded!
```

so maybe the site which you are navigating to blocked your IP?

**Comment by huaracheguarache** [comment]
@mxschmitt Thanks for taking a look at this! I just ran the container using Docker myself, and it succeeded like in your case. I suspected that it might have had something to do with Docker not having IPv6 access, but after disabling IPv6 on my machine a re-running the Podman container I got the same timeout result. So, it looks like the issue is caused by Podman in some way, and I just have to figure out why...
